geopandas/array.py

```diff
@@ -67,6 +68,20 @@ def _isna(value):
         return False
 
 
+def _check_crs(left, right, allow_none=False):
+    """
+    Check if the projection of both arrays is the same.
+
+    If allow_none is True, empty CRS is treated as the same.
+    """
+    if allow_none:
```

Comment:
The allow_none is not used at the moment?

Reply:
It is not, but we can decide to use it. I am just mimicking existing behaviour where the check is done as `if this.crs != other.crs:`. I would say that if we raise warning, it is okay to keep stricter condition. If we raised an error, I would be more permissive in the condition. I included it mostly for the potential future use.

Reply:
Fine to keep it